### PR TITLE
Enable GitHub Actions on the 3.9 branch

### PR DESCRIPTION
**Pull request by hugovk**
For https://github.com/python/core-workflow/issues/369, no bpo-NNNN available.

~Also update to v2 actions where available.~ --> Moved to own PR:  https://github.com/python/cpython/pull/20232.

### CODE REVIEW

File: .github/workflows/build_msi.yml [hunk 1]
```diff
@@ -21,14 +23,14 @@ jobs:
     name: 'Windows (x86) Installer'
     runs-on: windows-latest
     steps:
-    - uses: actions/checkout@v1
```
Comment (vstinner) [inline .github/workflows/build_msi.yml]: Can you please revert these changes and write a separated PR?

File: .github/workflows/doc.yml [hunk 2]
```diff
@@ -22,7 +24,7 @@ jobs:
     name: 'Docs'
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
```
Comment (vstinner) [inline .github/workflows/doc.yml]: Same remark.

**Comment by vstinner** [comment]
Oh, the "CPython Tests" job of Travis CI failed with network error on installing Debian packages:

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/bionic/main/binary-i386/Packages  Could not connect to archive.ubuntu.com:80 (91.189.88.152), connection timed out Could not connect to archive.ubuntu.com:80 (91.189.88.142), connection timed out [IP: 91.189.88.142 80]

File: .github/workflows/build_msi.yml (hunk 1 above)
Comment (hugovk) [reply .github/workflows/build_msi.yml]: Done!

Comment (hugovk) [reply .github/workflows/build_msi.yml]: And new PR: https://github.com/python/cpython/pull/20232

**Comment by vstinner** [comment]
Merged, thanks @hugovk.

**Comment by vstinner** [comment]
Modifying the master branch doesn't seem to be enough: https://github.com/python/cpython/pull/20228 didn't run GHA jobs after I rebased my PR to re-trigger all CI jobs.

I backported the PR to 3.9.
